File: b_rank_skillcheck_archive/concentration_04.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] hwn = ReadLine().Split().Select( int.Parse ).ToArray();
        int[][] t = new int[hwn[0]][];

        foreach( int i in Range( 0, hwn[0] ) ) {
            t[i] = ReadLine().Split().Select( int.Parse ).ToArray();
        }

        int p = int.Parse( ReadLine() );
        int[] ab = ReadLine().Split().Select( x => int.Parse( x ) - 1 ).ToArray();

        if( t[ab[0]][ab[1]] == t[ab[2]][ab[3]] ) {
            WriteLine( $"YES{System.Environment.NewLine}{p}" );
        } else {
            WriteLine( "NO" );
            WriteLine( hwn[2] < p + 1 ? 1 : p + 1 );
        }
        return;
    }
}

File: b_rank_skillcheck_archive/concentration_04.py
# 実行時間 10ms
h, w, n = list(map(int, input().split()))
t = [[0 for _ in [0] * w] for _ in [0] * h]

for i in range(h):
    t[i] = list(map(int, input().split()))

p = int(input())
a1, b1, a2, b2 = [int(x) - 1 for x in input().split()]

if t[a1][b1] == t[a2][b2]:
    print(f"YES\n{p}")
else:
    print("NO")
    print(1 if n < p + 1 else p + 1)
